src/index.ts

```diff
@@ -19,9 +19,19 @@ interface Config {
 // TODO: we need to give a way to consumers to ensure this does not
 // run multiple times on a single page load.
 const run = async (config: Config): Promise<void> => {
-  if (config.omitGdprConsent !== true) {
-    const hasConsent = await hasGdprConsent(config.vendorIds);
-    if (!hasConsent) return;
+  if (!config.omitGdprConsent) {
+    const { eventStatus, hasConsent } = await checkConsentStatus(
+      config.vendorIds
+    );
+
+    if (
+      (eventStatus === 'tcloaded' || eventStatus === 'useractioncomplete') &&
+      !hasConsent
+    ) {
+      return;
+    } else if (eventStatus === 'cmpuishown') {
+      await waitOnConsent(config.vendorIds, config.omitGdprConsent);
```

Comment:
This should not wait until consent is true - it should wait until consent is either true or false.